omc3/correction/handler.py

```diff
@@ -92,7 +92,18 @@ def correct(accel_inst: Accelerator, opt: DotDict) -> None:
                 # please look away for the next two lines.
                 accel_inst._model = corr_model
                 accel_inst._elements = corr_model_elements
-                resp_dict = response_twiss.create_response(accel_inst, opt.variable_categories, optics_params)
+
+                #If we are to compute the response including the DPP, then we have to do so from MAD-X (We do not have the analytical formulae), 
+                # otherwise we go through the other way of computing the response.
+                if DELTAP_NAME in opt.variable_categories:
+                    # Update dpp and create response around this dpp, then reset dpp (So _create_corrected_model is unaffected)
+                    old_dpp = accel_inst.dpp
+                    accel_inst.dpp = delta[DELTA][DELTAP_NAME]
+                    resp_dict = response_madx.create_fullresponse(accel_inst, opt.variable_categories)
+                    accel_inst.dpp = old_dpp
+                else:
+                    resp_dict = response_twiss.create_response(accel_inst, opt.variable_categories, optics_params)
+
```

Comment:
I don't like this whole part here, the loop is already convoluted enough. Maybe extract into a function. But also a few bugs here:
- DELTAP_NAME should be checked if in vars_list, not variable_categories
- I think the `accel_inst.dpp` should be `accel_inst.dpp = accel_inst.dpp + delta` (that's why it is a delta)
- I don't like how intransparent this switch from twiss to madx response is, there should be at least a LOG.info or LOG.warning for the user